### PR TITLE
faster like button search for image liking

### PR DESCRIPTION
**Pull request by douwevandermeij**
First check if there is a like or unlike button via existing spans, next find the like button if it exists, which is much faster.

### CODE REVIEW

**Comment by timgrossmann** [comment]
@douwevandermeij Looks good, thank you very much for the contribution!

Let's just wait for one more approval before we merge this.

File: instapy/like_util.py [hunk 1]
```diff
@@ -543,8 +543,14 @@ def check_link(browser,
...
-        "//a[@role='button']/span[text()='Like']/..")
+    like_elem = []
+
+    # fetch spans fast
+    spans = [x.text.lower() for x in browser.find_elements_by_xpath("//article//a[@role='button']/span")]
+
+    if 'like' in spans:
+        like_elem = browser.find_elements_by_xpath(
+            "//a[@role='button']/span[text()='Like']/..")
     if len(like_elem) == 1:
```
Comment (Tjorriemorrie) [inline instapy/like_util.py]: can remove this line then. and therefore also just remove `like_elem = []`

**Review by Tjorriemorrie — CHANGES_REQUESTED** [review]

File: instapy/like_util.py (hunk 1 above)
Comment (douwevandermeij) [reply instapy/like_util.py]: agree

**Comment by timgrossmann** [comment]
@douwevandermeij Awesome neat and concise upgrades for InstaPy, I love them!
Thank you very much for your contributions!

**Comment by douwevandermeij** [comment]
You're welcome!

**Pull request merged**